bundle/jsky.elevation.plot/src/main/java/jsky/plot/ElevationPanel.java

```diff
@@ -588,28 +572,48 @@ public void setItemColors(Paint[] colors) {
 
 
     // Add a marker to the given plot, showing the area of darkness
-    private void addDomainMarker(XYPlot xyPlot, Date startDate, Date endDate, float alpha, Color color) {
+    private void addDomainMarker(final XYPlot xyPlot,
+                                 final Date startDate, final String startDateName,
+                                 final Date endDate,   final String endDateName,
+                                 float alpha, Color color) {
+        final DateFormat df = ((DateAxis)xyPlot.getDomainAxis()).getDateFormatOverride();
         double start = startDate.getTime();
         double end = endDate.getTime();
 
         if (start < end) {
             IntervalMarker m = new IntervalMarker(start, end, color, DARKNESS_STROKE,
                     color, DARKNESS_STROKE, alpha);
-            xyPlot.addDomainMarker(m);
+            xyPlot.addDomainMarker(m, Layer.BACKGROUND);
+            xyPlot.addAnnotation(createAnnotation(String.format("%s: %s", startDateName, df.format(startDate)), start));
+            xyPlot.addAnnotation(createAnnotation(String.format("%s: %s", endDateName, df.format(endDate)), end));
         } else {
             double first = _model.getDateForHour(0.).getTime();
             double last = _model.getDateForHour(24.).getTime();
             IntervalMarker m1 = new IntervalMarker(start, last, color, DARKNESS_STROKE,
                     color, DARKNESS_STROKE, alpha);
-            xyPlot.addDomainMarker(m1);
+            xyPlot.addDomainMarker(m1, Layer.BACKGROUND);
+            xyPlot.addAnnotation(createAnnotation(String.format("%s: %s", startDateName, df.format(startDate)), start));
             IntervalMarker m2 = new IntervalMarker(first, end, color, DARKNESS_STROKE,
                     color, DARKNESS_STROKE, alpha);
-            xyPlot.addDomainMarker(m2);
+            xyPlot.addDomainMarker(m2, Layer.BACKGROUND);
+            xyPlot.addAnnotation(createAnnotation(String.format("%s: %s", endDateName, df.format(endDate)), first));
```

Comment:
The labels were added using `addAnnotation`, which was widely recommended on `JFreeChart` discussions online.